### PR TITLE
JSAB2-82 Create searchbar component

### PR DESCRIPTION
**Pull request by GuJialu**

### CODE REVIEW

**Review by HanqingZ — APPROVED** [review]

**Review by HONGTAO-T — APPROVED** [review]

File: frontend/src/Game/index.jsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import React from 'react';
 import { View } from 'react-native';
 import styles from '../common/styles';
 import Menu from '../Menu';
-import Buildings from '../Buildings';
+import Leaderboard from '../Leaderboard/index';
```
Comment (kyya) [inline frontend/src/Game/index.jsx]:
```suggestion
import Leaderboard from '../Leaderboard';
```
actually we don't need this index here.

**Review by kyya — APPROVED** [review]

File: frontend/src/Leaderboard/index.jsx [hunk 2]
```diff
@@ -20,9 +22,29 @@ const mockedUsers = [
...
+    flexDirection: 'row',
+    margin: 2,
+  },
+});
+
 function Leaderboard() {
+  function mockedSearchUser(username) {
+    // eslint-disable-next-line no-alert
+    alert(`search user ${username}`);
+  }
```
Comment (evasimonyi) [inline frontend/src/Leaderboard/index.jsx:43]: why do you put an alert here? just for testing?

File: frontend/src/Leaderboard/index.jsx [hunk 3]
```diff
@@ -20,9 +22,29 @@ const mockedUsers = [
...
+
   return (
     <ScrollView>
+      <View style={styles.searchBarView}>
+        <SearchBar style={styles.searchBar} onSubmit={mockedSearchUser} />
```
Comment (evasimonyi) [inline frontend/src/Leaderboard/index.jsx]: when I click inside the search bar, I think the best user experience is if the icon disappears
refactor the styling, it's too narrow, very hard to click into it, make it a little bigger
also, you can use some placeholder text as well, like 'Search for users' or something

please create a new component for the search bar, we might use it elsewhere in the app!

**Review by evasimonyi — CHANGES_REQUESTED** [review]

File: frontend/src/Leaderboard/index.jsx (hunk 2 above)
Comment (GuJialu) [reply frontend/src/Leaderboard/index.jsx:43]: Yes, just to show the submit function works.

File: frontend/src/Leaderboard/index.jsx (hunk 3 above)
Comment (GuJialu) [reply frontend/src/Leaderboard/index.jsx]: By saying refactor the styling, does it mean we should enlarge the search bar when the user clicks on it?

Comment (evasimonyi) [reply frontend/src/Leaderboard/index.jsx]: I think it should be bigger by default because it's so small it's hard to click in it on my phone.

**Review by kyya — APPROVED** [review]

**Review by HONGTAO-T — APPROVED** [review]

File: frontend/src/Leaderboard/index.jsx [hunk 4]
```diff
@@ -20,9 +22,31 @@ const mockedUsers = [
   },
 ];
 
+const styles = StyleSheet.create({
+  searchBar: {
+    width: '100%',
+    fontSize: 24,
+    borderRadius: 10,
+    backgroundColor: Colors.lightenGrey,
```
Comment (evasimonyi) [inline frontend/src/Leaderboard/index.jsx:30]: please add some margin and padding as we discussed

File: frontend/src/common/components/SearchBar.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import { TextInput } from 'react-native';
+import PropTypes from 'prop-types';
+
+const defaultStyle = {
+  fontSize: 24,
+};
+
+function SearchBar({ style, onSubmit, placeholder }) {
+  const [val, setVal] = useState('');
```
Comment (evasimonyi) [inline frontend/src/common/components/SearchBar.jsx:10]:
```suggestion
  const [val, setVal] = useState('');
  
```
for better readability add a line

**Review by evasimonyi — CHANGES_REQUESTED** [review]

**Review by evasimonyi — APPROVED** [review]

**Pull request merged**